docker/build.sh

```diff
@@ -2,76 +2,135 @@
 
 set -e
 
+# Function to print help message
+print_help() {
+    echo "Usage: build.sh [OPTIONS]"
+    echo "Options:"
+    echo "  --help          Display this help message"
+    echo "  -h              Display this help message"
+    echo "  --no-cuda       Disable CUDA support"
+    echo "  --platform      Specify the platform (default: current platform)"
+    echo "  --devel-only    Build devel image only"
+    echo ""
+    echo "Note: The --platform option should be one of 'linux/amd64' or 'linux/arm64'."
```

Comment:
I know that the `--platform` option existed before this PR, but testing it now for the first time I can't cross-compile to aarch64 on an x86_64 machine, it fails during the compilation of Autoware with
```
#26 2700.1 --- stderr: ament_cmake_gen_version_h                                                                                                                                                           
#26 2700.1 CMake Error at /usr/share/cmake-3.22/Modules/CMakeDetermineCCompiler.cmake:49 (message):                                                                                                        
#26 2700.1   Could not find compiler set in environment variable CC:                                                                                                                                       
#26 2700.1                                                                                                                                                                                                 
#26 2700.1   /usr/lib/ccache/gcc.                                                                                                                                                                          
#26 2700.1 Call Stack (most recent call first):                                                                                                                                                            
#26 2700.1   CMakeLists.txt:2 (project)                                                                                                                                                                    
#26 2700.1                                                                                                                                                                                                 
#26 2700.1                                                                                                                                                                                                 
#26 2700.1 CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage                                                                                                                                     
#26 2700.1 CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage                                                                                                                                   
#26 2700.1 ---                                                                                                                                                                                             
#26 2700.1 Failed   <<< ament_cmake_gen_version_h [3.48s, exited with code 1]
```

I don't know if we currently make use of this flag anywhere (CI for example), or even if it was working before this PR. If it was working it should probably be fixed. If it was not working we may want to consider if it is a good idea to keep an option that doesn't work.

Reply:
@mitsudome-r do you know anything about this flag?

If not used, we can remove it in a consecutive PR.

I will create an issue for it.

Reply:
@ambroise-arm This was about ccache update, now it should be resolved can you pull the branch again ?

Reply:
https://github.com/autowarefoundation/autoware/issues/4522

Reply:
I can at least say it now passes the point where I saw the reported failure (still cross-compiling, it is very slow). I can create a new thread if I see something else. (can be resolved)